lib/internal/test_runner/fake_timers.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+
+const {
+  DateNow,
+  SafeMap,
+  Symbol,
+  globalThis,
+} = primordials;
+
+class Timers {
+  constructor() {
+    this.timers = new SafeMap();
+
+    this.setTimeout = this.#createTimer.bind(this, false);
+    this.clearTimeout = this.#clearTimer.bind(this);
+    this.setInterval = this.#createTimer.bind(this, true);
+    this.clearInterval = this.#clearTimer.bind(this);
+  }
+
+  #createTimer(isInterval, callback, delay, ...args) {
+    const timerId = Symbol('kTimerId');
+    const timer = {
+      id: timerId,
+      callback,
+      time: DateNow() + delay,
```

Comment:
nit: I like Sinon's naming `timer.runAt` better than `timer.time` for clarity

Reply:
perfect!

Reply:
done!